recipe/common.php

```diff
@@ -127,7 +140,18 @@
         $at = "-b $branch";
     }
 
-    run("git clone $at --depth 1 --recursive -q $repository {{release_path}} 2>&1");
+    $releases = env('releases_list');
+    
+    if($gitCache && isset($releases[1])){
+      try {
+       run("git clone $at --recursive -q --reference {{deploy_path}}/releases/{$releases[1]} --dissociate $repository  {{release_path}} 2>&1");  
+      } catch (RuntimeException $exc) {
+        // If {{deploy_path}}/releases/{$releases[1]} has a failed git clone, is empty, shallow etc, git would throw error and give up. So we're forcing it to act without reference in this situation
+        run("git clone $at --recursive -q $repository {{release_path}} 2>&1"); 
+      }
+    } else{
+      run("git clone $at --depth 1 --recursive -q $repository {{release_path}} 2>&1"); 
```

Comment:
It's not duplicate code: this line creates shallow clone. Shallow clone cannot be used as a reference point. Also clone with full history takes more time to do than shallow one.

So reasoning is this:
1. We have git supporting reference and dissociate, and we have possible reference
  - try clonig with reference
  - well, that didn't work out, do normal clone
2. Well we don't have git supporting/dont want to use it, so clone shallow.

point 2 fails under one condition: initial git cloning, without first reference, would be shallow clone. next cloning would fail to reference, so we get full clone and then subsequent cloning would be OK... I think I had this figured right at some point in local common, let me check.